### ISSUE TITLE
How does custom configuration actually work?

### ISSUE THREAD

**Issue by originalnoe-nazvanie** [instaclustr/cassandra-operator#335]
I tried to follow the instruction on
https://github.com/instaclustr/cassandra-operator/blob/master/doc/custom-configuration.md
to pass the test conf sample to Cassabdra cluster, but it seems that something go wrong.

1) on https://github.com/instaclustr/cassandra-operator/blob/master/doc/custom-configuration.md apiVersion is stable.instaclustr.com/v1, 
but on 
https://github.com/instaclustr/cassandra-operator/blob/master/examples/example-datacenter.yaml apiVersion: cassandraoperator.instaclustr.com/v1alpha1
2) when i try to apply cluster conf, containing 
```yaml
  userConfigMapVolumeSource:
    name: concurrent-data
    items:
      - key: 100-concurrent-yaml
        path: cassandra.yaml.d/100-concurrent.yaml
```
i get
```
PS C:\Projects\Sources\cassandra-aks> .\kubectl.exe apply -f cluster.yml   
The CassandraDataCenter "dev-cassandra" is invalid: []: Invalid value: map[string]interface {}{"apiVersion":"cassandraoperator.instaclustr.com/v1alpha1", "kind":"CassandraDataCenter", "metadata":map[string]interface {}{"annotations":map[string]interface {}{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"cassandraoperator.instaclustr.com/v1alpha1\",\"kind\":\"CassandraDataCenter\",\"metadata\":{\"annotations\":{},\"labels\":{\"app\":\"cassandra\"},\"name\":\"dev-cassandra\",\"namespace\":\"default\"},\"spec\":{\"backup\":{\"backupName\":\"test-cassandra-backup-restore-s3\",\"backupSecretVolumeSource\":{\"secretName\":\"backup-secrets\"},\"restore\":false},\"cassandraImage\":\"gcr.io/cassandra-operator/cassandra:3.11.5\",\"dataVolumeClaimSpec\":{\"accessModes\":[\"ReadWriteOnce\"],\"resources\":{\"requests\":{\"storage\":\"2Gi\"}},\"storageClassName\":\"standard\"},\"fsGroup\":999,\"nodes\":2,\"operatorLabels\":{\"nodesService\":{\"mynodesservicelabel\":\"nodelabel\"},\"podTemplate\":{\"mypodlabel\":\"podlabel\"},\"prometheusService\":{\"customPrometheusLabel\":\"prometheuslabel\"},\"seedNodesService\":{\"myseedlabel\":\"seednode\"},\"statefulSet\":{\"mystatefullabel\":\"statefulsetlabel\"}},\"optimizeKernelParams\":true,\"prometheusSupport\":true,\"resources\":{\"limits\":{\"memory\":\"1Gi\"},\"requests\":{\"memory\":\"1Gi\"}},\"serviceAccountName\":\"cassandra-performance\",\"sidecarEnv\":[{\"name\":\"AZURE_STORAGE_ACCOUNT\",\"valueFrom\":{\"secretKeyRef\":{\"key\":\"azurestorageaccount\",\"name\":\"backup-secrets\"}}},{\"name\":\"AZURE_STORAGE_KEY\",\"valueFrom\":{\"secretKeyRef\":{\"key\":\"azurestoragekey\",\"name\":\"backup-secrets\"}}}],\"sidecarImage\":\"gcr.io/cassandra-operator/cassandra-sidecar:latest\",\"sidecarResources\":{\"limits\":{\"memory\":\"512Mi\"},\"requests\":{\"memory\":\"512Mi\"}},\"userConfigMapVolumeSource\":{\"items\":[{\"key\":\"100-concurrent-yaml\",\"path\":\"cassandra.yaml.d/100-concurrent.yaml\"}],\"name\":\"concurrent-data\"}}}\n"}, "creationTimestamp":"2020-01-27T10:13:35Z", "generation":13, "labels":map[string]interface {}{"app":"cassandra"}, "name":"dev-cassandra", "namespace":"default", "resourceVersion":"111355", "uid":"ada289f7-40ed-11ea-89c2-fa6ad152a9e8"}, "spec":map[string]interface {}{"backup":map[string]interface {}{"backupName":"test-cassandra-backup-restore-s3", "backupSecretVolumeSource":map[string]interface {}{"secretName":"backup-secrets"}, "restore":false}, "cassandraImage":"gcr.io/cassandra-operator/cassandra:3.11.5", "dataVolumeClaimSpec":map[string]interface {}{"accessModes":[]interface {}{"ReadWriteOnce"}, "resources":map[string]interface {}{"requests":map[string]interface {}{"storage":"2Gi"}}, "storageClassName":"standard"}, "fsGroup":999, "imagePullPolicy":"IfNotPresent", "nodes":2, "operatorLabels":map[string]interface 
{}{"nodesService":map[string]interface {}{"mynodesservicelabel":"nodelabel"}, "podTemplate":map[string]interface {}{"mypodlabel":"podlabel"}, "prometheusService":map[string]interface {}{"customPrometheusLabel":"prometheuslabel"}, "seedNodesService":map[string]interface {}{"myseedlabel":"seednode"}, "statefulSet":map[string]interface {}{"mystatefullabel":"statefulsetlabel"}}, "optimizeKernelParams":true, "prometheusSupport":true, "resources":map[string]interface {}{"limits":map[string]interface {}{"memory":"1Gi"}, "requests":map[string]interface {}{"memory":"1Gi"}}, "serviceAccountName":"cassandra-performance", "sidecarEnv":[]interface {}{map[string]interface {}{"name":"AZURE_STORAGE_ACCOUNT", "valueFrom":map[string]interface {}{"secretKeyRef":map[string]interface {}{"key":"azurestorageaccount", "name":"backup-secrets"}}}, map[string]interface {}{"name":"AZURE_STORAGE_KEY", "valueFrom":map[string]interface {}{"secretKeyRef":map[string]interface {}{"key":"azurestoragekey", "name":"backup-secrets"}}}}, "sidecarImage":"gcr.io/cassandra-operator/cassandra-sidecar:latest", "sidecarResources":map[string]interface {}{"limits":map[string]interface {}{"memory":"512Mi"}, "requests":map[string]interface {}{"memory":"512Mi"}}, "userConfigMapVolumeSource":map[string]interface {}{"items":[]interface {}{map[string]interface {}{"key":"100-concurrent-yaml", "path":"cassandra.yaml.d/100-concurrent.yaml"}}, "name":"concurrent-data"}}}: validation failure list:
type in spec.userConfigMapVolumeSource is required
```
3) if i try to apply
```yaml
  userConfigMapVolumeSource:
    name: concurrent-data
    type: array
    items:
      - key: 100-concurrent-yaml
        path: cassandra.yaml.d/100-concurrent.yaml
``` 		
it applyes, but nowhing happens
4) Moreover, if i try to apply cluster manifest containing 
```yaml
  userConfigMapVolumeSource:
    name: concurrent-data
    type: array
    items:
      - key: 100-concurrent-yaml
        path: cassandra.yaml.d/100-concurrent.yaml
```
when no cluster exists, nothing happens at all, cluster is not creating

I'll be really happy if you could show me the right direction =)

**Comment by smiklosovic** [comment]
hey @originalnoe-nazvanie , can you please show me the complete yaml you are applying?

The `concurrent-data` is name of a `ConfigMap` you need to have already present in your Kubernetes cluster and it has to have entry "100-concurrent-yaml", its content will be then mounted uner `/etc/cassandra/casandra.yaml.d/100-concurrent.yaml`.

I am not sure why "nothing is happening", if you have your operator running and changes are applied, you should see some logging message about that in operator's output.

If you apply the configuration on already running cluster, you will need to restart that pod (or whole cluster, in round robin fashion), so next start of a pod will "re-fetch" the config map you have just applied.

In other words, if you apply some custom configuration on already running cluster, in order to see these changes, you need to practically restart a Cassandra node (hence pod) to see it in effect. Cassandra itself can not reconfigure itself just because you have mounted a ConfigMap somewhere - restart of that pod will re-read and re-apply these changes.

**Comment by originalnoe-nazvanie** [comment]
Hi! Appreciate your help, thank you!
The configmap present in cluster and I already tried to kill the cassandra pod, but the problem appears even if 
delete the cluster and create it again, no pods get started.
There is a hint with fsGroup: 999 on
https://github.com/instaclustr/cassandra-operator/blob/master/doc/providers/aks.md
Is it possible that it's Azure (AKS)-specific problem?

```yaml
PS C:\Projects\Sources\cassandra-aks> .\kubectl.exe describe configmap concurrent-data
Name:         concurrent-data
Namespace:    default
Labels:       <none>
Annotations:  <none>


Data
====
100-concurrent-yaml:
----
concurrent_reads: 12
concurrent_writes: 12
concurrent_counter_writes: 12
Events:  <none>
```
Full cluster yaml
```yaml
apiVersion: cassandraoperator.instaclustr.com/v1alpha1
kind: CassandraDataCenter
metadata:
  name: dev-cassandra
  labels:
    app: cassandra
spec:
  nodes: 2
  cassandraImage: "gcr.io/cassandra-operator/cassandra:3.11.5"
  sidecarImage: "gcr.io/cassandra-operator/cassandra-sidecar:latest"

  backup:
    restore: false
    backupName: test-cassandra-backup-restore-s3
    backupSecretVolumeSource:
      secretName: backup-secrets

  sidecarEnv:
    - name: AZURE_STORAGE_ACCOUNT
      valueFrom:
        secretKeyRef:
          name: backup-secrets
          key: azurestorageaccount
    - name: AZURE_STORAGE_KEY
      valueFrom:
        secretKeyRef:
          name: backup-secrets
          key: azurestoragekey
  resources:
    limits:
      memory: 1Gi
    requests:
      memory: 1Gi
  sidecarResources:
    limits:
      memory: 512Mi
    requests:
      memory: 512Mi

  dataVolumeClaimSpec:
    accessModes:
      - ReadWriteOnce
    storageClassName: standard
    resources:
      requests:
        storage: 2Gi

  operatorLabels:
    nodesService:
      mynodesservicelabel: nodelabel
    seedNodesService:
      myseedlabel: seednode
    statefulSet:
      mystatefullabel: statefulsetlabel
    podTemplate:
      mypodlabel: podlabel
    prometheusService:
      customPrometheusLabel: prometheuslabel

  prometheusSupport: true
  optimizeKernelParams: true
  serviceAccountName: cassandra-performance
  fsGroup: 999

# that doesn't work =(
#  userConfigMapVolumeSource:
#    name: concurrent-data
#    type: array
#    items:
#      - key: 100-concurrent-yaml
#        path: cassandra.yaml.d/100-concurrent.yaml
```

could you help with troubleshooting the operator logs?

**Comment by smiklosovic** [comment]
Most probably, your cluster is waiting on something to happen so it can proceed.

What does `kubectl describe cassandradatacenters<tab> <tab>` gives you?

Try to gather as much logs from describe command as possible ...

**Comment by smiklosovic** [comment]
@originalnoe-nazvanie 
I think I know what your issue is, if you have a file called "custom-cassandra-config.yaml" with this content:

```
concurrent_reads: 12
concurrent_writes: 12
concurrent_counter_writes: 12
```

And you create a ConfigMap with this command:

```
kubectl create configmap concurrent-data --from-file=custom-cassandra-config.yaml
```

Then you have it like this:

```
$ kubectl describe configmap concurrent-data 
Name:         concurrent-data
Namespace:    default
Labels:       <none>
Annotations:  <none>

Data
====
custom-cassandra-config.yaml:     <--- you see this?
----
concurrent_reads: 12
concurrent_writes: 12
concurrent_counter_writes: 12
```

So you need to use that as key in CDC spec:

```
  userConfigMapVolumeSource:
    name: concurrent-data
    type: array
    items:
      - key: custom-cassandra-config.yaml       <--- this has to match what you have in configmap
        path: cassandra.yaml.d/100-concurrent.yaml

```

**Comment by originalnoe-nazvanie** [comment]
1)I recreated a configmap with settings, provided by you

```yaml
PS C:\Projects\Sources\cassandra-aks> kubectl delete configmap concurrent-data
configmap "concurrent-data" deleted
PS C:\Projects\Sources\cassandra-aks> kubectl create configmap concurrent-data --from-file=custom-cassandra-config.yaml
configmap/concurrent-data created
PS C:\Projects\Sources\cassandra-aks> kubectl describe configmap concurrent-data                                       
Name:         concurrent-data
Namespace:    default
Labels:       <none>
Annotations:  <none>

Data
====
custom-cassandra-config.yaml:
----
concurrent_reads: 12
concurrent_writes: 12
concurrent_counter_writes: 12
Events:  <none>
```
2) uncomment the section
```yaml                                      
  userConfigMapVolumeSource:
    name: concurrent-data
    type: array
    items:
      - key: custom-cassandra-config.yaml
        path: cassandra.yaml.d/100-concurrent.yaml
```
3) changed a base image
```yaml 
spec:
  nodes: 2
  cassandraImage: "gcr.io/cassandra-operator/cassandra-3.11.5:latest"
  sidecarImage: "gcr.io/cassandra-operator/cassandra-sidecar:latest"
```

but again, no success =(

```yaml
PS C:\Projects\Sources\cassandra-aks> .\kubectl.exe describe cassandradatacenters
Name:         dev-cassandra
Namespace:    default
Labels:       app=cassandra
Annotations:  kubectl.kubernetes.io/last-applied-configuration:
                {"apiVersion":"cassandraoperator.instaclustr.com/v1alpha1","kind":"CassandraDataCenter","metadata":{"annotations":{},"labels":{"app":"cass...
API Version:  cassandraoperator.instaclustr.com/v1alpha1
Kind:         CassandraDataCenter
Metadata:
  Creation Timestamp:  2020-01-28T10:15:03Z
  Generation:          1
  Resource Version:    222779
  Self Link:           /apis/cassandraoperator.instaclustr.com/v1alpha1/namespaces/default/cassandradatacenters/dev-cassandra
  UID:                 0c762619-41b7-11ea-89c2-fa6ad152a9e8
Spec:
  Backup:
    Backup Name:  test-cassandra-backup-restore-s3
    Backup Secret Volume Source:
      Secret Name:  backup-secrets
    Restore:        false
  Cassandra Image:  gcr.io/cassandra-operator/cassandra-3.11.5:latest
  Data Volume Claim Spec:
    Access Modes:
      ReadWriteOnce
    Resources:
      Requests:
        Storage:         2Gi
    Storage Class Name:  standard
  Fs Group:              999
  Nodes:                 2
  Operator Labels:
    Nodes Service:
      Mynodesservicelabel:  nodelabel
    Pod Template:
      Mypodlabel:  podlabel
    Prometheus Service:
      Custom Prometheus Label:  prometheuslabel
    Seed Nodes Service:
      Myseedlabel:  seednode
    Stateful Set:
      Mystatefullabel:     statefulsetlabel
  Optimize Kernel Params:  true
  Prometheus Support:      true
  Resources:
    Limits:
      Memory:  1Gi
    Requests:
      Memory:            1Gi
  Service Account Name:  cassandra-performance
  Sidecar Env:
    Name:  AZURE_STORAGE_ACCOUNT
    Value From:
      Secret Key Ref:
        Key:   azurestorageaccount
        Name:  backup-secrets
    Name:      AZURE_STORAGE_KEY
    Value From:
      Secret Key Ref:
        Key:      azurestoragekey
        Name:     backup-secrets
  Sidecar Image:  gcr.io/cassandra-operator/cassandra-sidecar:latest
  Sidecar Resources:
    Limits:
      Memory:  512Mi
    Requests:
      Memory:  512Mi
  User Config Map Volume Source:
    Items:
      Key:   custom-cassandra-config.yaml
      Path:  cassandra.yaml.d/100-concurrent.yaml
    Name:    concurrent-data
    Type:    array
Events:      <none>
```

```yaml
NAME                                 READY   STATUS    RESTARTS   AGE
cassandra-operator-77b977f47-mf6sf   1/1     Running   0          23h
```

**Comment by smiklosovic** [comment]
Go with minimal Spec, you have too much specified, you do not need that. It most probably waits for all secrets and so on and you do not have them so it is blocked.

what does decsribe does on that cassandradatacenter resource?

Try e.g this and look into logs `kubectl describe statefulsets`

```
apiVersion: cassandraoperator.instaclustr.com/v1alpha1
kind: CassandraDataCenter
metadata:
  name: test-dc-cassandra
spec:
  cassandraImage: "gcr.io/cassandra-operator/cassandra-3.11.5:latest"
  sidecarImage: "gcr.io/cassandra-operator/cassandra-sidecar:latest"
  nodes: 2
  dataVolumeClaimSpec:
    accessModes:
      - ReadWriteOnce
    resources:
      requests:
        storage: 1Gi
  resources:
    limits:
      memory: 1Gi 
    requests:
      memory: 1Gi 
  sidecarResources:
    limits:
      memory: 512Mi
    requests:
      memory: 512Mi
```

**Comment by originalnoe-nazvanie** [comment]
The configuration with backup, azure storage_account and PVC works, the only problem is userConfigMapVolumeSource =(
1) applying manifest with 
```yaml
  userConfigMapVolumeSource:
    name: concurrent-data
    type: array
    items:
      - key: 100-concurrent-yaml
        path: cassandra.yaml.d/100-concurrent.yaml
```
2) i manually delete statefulset and watched over pods for logs
```yaml
PS C:\Projects\Sources\cassandra-aks> kubectl describe pods cassandra-dev-cassandra-rack1-0
Name:               cassandra-dev-cassandra-rack1-0
Namespace:          default
Priority:           0
PriorityClassName:  <none>
Node:               aks-nodepool1-40738392-0/10.240.0.5
Start Time:         Tue, 28 Jan 2020 16:57:40 +0300
Labels:             app.kubernetes.io/managed-by=com.instaclustr.cassandra-operator
                    cassandra-operator.instaclustr.com/datacenter=dev-cassandra
                    cassandra-operator.instaclustr.com/rack=rack1
                    controller-revision-hash=cassandra-dev-cassandra-rack1-7b477fbbd6
                    mypodlabel=podlabel
                    statefulset.kubernetes.io/pod-name=cassandra-dev-cassandra-rack1-0
Annotations:        <none>
Status:             Pending
IP:
Controlled By:      StatefulSet/cassandra-dev-cassandra-rack1
Init Containers:
  sysctl-limits:
    Container ID:
    Image:         busybox:latest
    Image ID:
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -xuec
    Args:
      sysctl -w vm.max_map_count=1048575
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from cassandra-performance-token-vsfxm (ro)
Containers:
  cassandra:
    Container ID:
    Image:         gcr.io/cassandra-operator/cassandra-3.11.5:latest
    Image ID:
    Ports:         7000/TCP, 7001/TCP, 9042/TCP, 7199/TCP, 9500/TCP
    Host Ports:    0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP
    Args:
      /tmp/operator-config
      /tmp/cassandra-rack-config
      /tmp/user-config
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Limits:
      memory:  1Gi
    Requests:
      memory:   1Gi
    Readiness:  exec [/usr/bin/cql-readiness-probe] delay=60s timeout=5s period=10s #success=1 #failure=3
    Environment:
      MEMORY_LOCK:  true
    Mounts:
      /tmp/cassandra-rack-config from operator-rack-config-volume-rack1 (rw)
      /tmp/operator-config from operator-config-volume (rw)
      /tmp/user-config from concurrent-data (rw)
      /var/lib/cassandra from data-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from cassandra-performance-token-vsfxm (ro)
  sidecar:
    Container ID:
    Image:          gcr.io/cassandra-operator/cassandra-sidecar:latest
    Image ID:
    Port:           4567/TCP
    Host Port:      0/TCP
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Limits:
      memory:  512Mi
    Requests:
      memory:  512Mi
    Environment:
      AZURE_STORAGE_ACCOUNT:  <set to the key 'azurestorageaccount' in secret 'backup-secrets'>  Optional: false
      AZURE_STORAGE_KEY:      <set to the key 'azurestoragekey' in secret 'backup-secrets'>      Optional: false
    Mounts:
      /etc/pod-info from pod-info (rw)
      /tmp/backup-secret from backup-secrets (rw)
      /var/lib/cassandra from data-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from cassandra-performance-token-vsfxm (ro)
Conditions:
  Type              Status
  Initialized       False
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  data-volume:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  data-volume-cassandra-dev-cassandra-rack1-0
    ReadOnly:   false
  pod-info:
    Type:  DownwardAPI (a volume populated by information about the pod)
    Items:
      metadata.labels -> labels
      metadata.annotations -> annotations
      metadata.namespace -> namespace
      metadata.name -> name
  operator-config-volume:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      cassandra-dev-cassandra-operator-config
    Optional:  false
  operator-rack-config-volume-rack1:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      cassandra-dev-cassandra-rack-config-rack1
    Optional:  false
  backup-secrets:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  backup-secrets
    Optional:    false
  concurrent-data:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      concurrent-data
    Optional:  false
  cassandra-performance-token-vsfxm:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  cassandra-performance-token-vsfxm
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason                  Age                 From                               Message
  ----     ------                  ----                ----                               -------
  Normal   Scheduled               2m13s               default-scheduler                  Successfully assigned default/cassandra-dev-cassandra-rack1-0 to aks-nodepool1-40738392-0
  Warning  FailedAttachVolume      2m13s               attachdetach-controller            Multi-Attach error for volume "pvc-baa371a2-4039-11ea-89c2-fa6ad152a9e8" Volume is already exclusively attached to one node and can't be attached to another
  Normal   SuccessfulAttachVolume  81s                 attachdetach-controller            AttachVolume.Attach succeeded for volume "pvc-baa371a2-4039-11ea-89c2-fa6ad152a9e8"
  Warning  FailedMount             10s                 kubelet, aks-nodepool1-40738392-0  Unable to mount volumes for pod "cassandra-dev-cassandra-rack1-0_default(26108991-41d6-11ea-89c2-fa6ad152a9e8)": timeout expired 
waiting for volumes to attach or mount for pod "default"/"cassandra-dev-cassandra-rack1-0". list of unmounted volumes=[concurrent-data]. list of unattached volumes=[data-volume pod-info operator-config-volume operator-rack-config-volume-rack1 backup-secrets concurrent-data cassandra-performance-token-vsfxm]
  Warning  FailedMount             5s (x9 over 2m13s)  kubelet, aks-nodepool1-40738392-0  MountVolume.SetUp failed for volume "concurrent-data" : configmap references non-existent config key: 100-concurrent-yaml   
```
  
so
```yaml
MountVolume.SetUp failed for volume "concurrent-data" : configmap references non-existent config key: 100-concurrent-yaml  
```
was the problem 
actual configs that works
```yaml
PS C:\Projects\Sources\cassandra-aks> kubectl get configmaps
NAME                                        DATA   AGE 
cassandra-dev-cassandra-operator-config     3      53m 
cassandra-dev-cassandra-rack-config-rack1   1      53m 
cassandra-operator-default-config           6      2d1h
cassandra-operator-lock                     0      27h 
concurrent-data                             1      12m 
PS C:\Projects\Sources\cassandra-aks> kubectl describe configmaps concurrent-data
Name:         concurrent-data
Namespace:    default        
Labels:       <none>
Annotations:  <none>

Data
====
custom-cassandra-config.yaml:
----
concurrent_reads: 12
concurrent_writes: 12        
concurrent_counter_writes: 12
Events:  <none>
```
```yaml
...
  userConfigMapVolumeSource:
    name: concurrent-data
    type: array
    items:
      - key: custom-cassandra-config.yaml
        path: cassandra.yaml.d/100-concurrent.yaml
...
```

Thank you for feedback, your operator looks cool!
